HIRS_AttestationCA/src/main/java/hirs/attestationca/AbstractAttestationCertificateAuthority.java

```diff
@@ -425,12 +430,13 @@ private IdentityResponseEnvelope generateIdentityResponseEnvelopeAndStoreIssuedC
      * Performs supply chain validation.
      *
      * @param claim the identity claim
+     * @param ekPub the public endorsement key
      * @return the {@link AppraisalStatus} of the supply chain validation
      */
     private AppraisalStatus.Status doSupplyChainValidation(
-            final ProvisionerTpm2.IdentityClaim claim) {
+            final ProvisionerTpm2.IdentityClaim claim, final PublicKey ekPub) {
```

Comment:
Yeah, trust me, I had considered that route, but we also use the ekPub right after this method completes in the calling method, thus instead of regenerating the key twice and wasting cycles, I figured we could generate it once in the calling method and use it in both places.

Reply:
Fair enough!